src/controllers/DynamicFacetQueryController.ts

```diff
@@ -63,23 +74,19 @@ export class DynamicFacetQueryController {
   public putFacetIntoQueryBuilder(queryBuilder: QueryBuilder) {
     Assert.exists(queryBuilder);
 
-    queryBuilder.facetRequests.push(this.facetRequest);
+    queryBuilder.facetRequests.push(this.buildFacetRequest(queryBuilder.build()));
     if (this.freezeFacetOrder) {
       queryBuilder.facetOptions.freezeFacetOrder = true;
     }
   }
 
-  public get facetRequest(): IFacetRequest {
+  public buildFacetRequest(query: IQuery): IFacetRequest {
     return {
-      facetId: this.facet.options.id,
-      field: this.facet.fieldName,
-      type: this.facet.facetType,
-      sortCriteria: this.facet.options.sortCriteria as FacetSortCriteria,
+      ...this.requestBuilder.buildBaseRequestForQuery(query),
```

Comment:
Should this line be at the end to prevent properties below from overwriting those in the desctructured object, or is overwriting the goal?

Reply:
It makes more sense at the beginning IMO. You're getting the baseRequest, then adding more specific properties. Normally you won't have any properties being repeated, but if it's the case the ones from bellow should have priority.